employee-rostering-frontend/src/ui/components/DataTable.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  headerCol,
+  ICell
+} from '@patternfly/react-table';
+
+interface DataTableProps {
+  title: string;
+  columnTitles: string[];
+  dataSupplier: (_: ((_: string[][]) => void)) => void;
+}
+
+interface DataTableState {
+  columns: ICell[];
+  data: string[][];
+  actions: { 'title': string; 'onClick': (event: any, rowId: number, rowData: string[], extra: any) => void }[];
+  currentFilter: (_: string[]) => boolean;
```

Comment:
The code would be easier to understand if all parameters were named (`rows` in this case).